src/core/PathUtil.cpp

```diff
@@ -0,0 +1,135 @@
+#include "PathUtil.h"
+
+#include <QDebug>
+#include <QDir>
+#include <QFileInfo>
+
+#include "ConfigManager.h"
+
+namespace PathUtil
+{
+	std::vector<Base> relativeBases = { ProjectDirBase, FactorySampleBase, UserSampleBase, UserVSTBase, PresetBase,
+		UserLADSPABase, DefaultLADSPABase, UserSoundfontBase, DefaultSoundfontBase, UserGIGBase, DefaultGIGBase };
+
+	QString baseLocation(Base base)
+	{
+		QString loc = "";
+		switch (base)
+		{
+			case ProjectDirBase       : loc = ConfigManager::inst()->userProjectsDir(); break;
+			case FactorySampleBase    :
+			{
+				QDir fsd = QDir(ConfigManager::inst()->factorySamplesDir());
+				loc = fsd.absolutePath() + "/"; break;
+			}
+			case UserSampleBase       : loc = ConfigManager::inst()->userSamplesDir(); break;
+			case UserVSTBase          : loc = ConfigManager::inst()->userVstDir(); break;
+			case PresetBase           : loc = ConfigManager::inst()->userPresetsDir(); break;
+			case UserLADSPABase       : loc = ConfigManager::inst()->ladspaDir(); break;
+			case DefaultLADSPABase    : loc = ConfigManager::inst()->userLadspaDir(); break;
+			case UserSoundfontBase    : loc = ConfigManager::inst()->sf2Dir(); break;
+			case DefaultSoundfontBase : loc = ConfigManager::inst()->userSf2Dir(); break;
+			case UserGIGBase          : loc = ConfigManager::inst()->gigDir(); break;
+			case DefaultGIGBase       : loc = ConfigManager::inst()->userGigDir(); break;
+			default                   : return QString("");
+		}
+		return QDir::cleanPath(loc) + QDir::separator();
+	}
+
+	QDir baseQDir (Base base) { return QDir(baseLocation(base)); }
+
+	QString basePrefix(Base base)
+	{
+		switch (base)
+		{
+			case ProjectDirBase       : return "userprojects:";
+			case FactorySampleBase    : return "factorysample:";
+			case UserSampleBase       : return "usersample:";
+			case UserVSTBase          : return "uservst:";
+			case PresetBase           : return "preset:";
+			case UserLADSPABase       : return "userladspa:";
+			case DefaultLADSPABase    : return "defaultladspa:";
+			case UserSoundfontBase    : return "usersoundfont:";
+			case DefaultSoundfontBase : return "defaultsoundfont:";
+			case UserGIGBase          : return "usergig:";
+			case DefaultGIGBase       : return "defaultgig:";
+			default                   : return "";
+		}
+	}
+
+	Base baseLookup(QString path)
+	{
+		for (auto base: relativeBases)
+		{
+			QString prefix = basePrefix(base);
+			if ( prefix == path.left(prefix.length()) ) { return base; }
```

Comment:
You may use [`QString::startWith()`](https://doc.qt.io/qt-5/qstring.html#startsWith) here.